proto/cosmos/base/reflection/v1beta1/reflection.proto

```diff
@@ -21,6 +25,187 @@ service ReflectionService {
   };
 }
 
+// AppDescriptor describes a cosmos-sdk based application
+message AppDescriptor {
+  // chain provides the chain descriptor
+  ChainDescriptor chain = 2;
+  // codec provides metadata information regarding codec related types
+  CodecDescriptor codec = 3;
+  // configuration provides metadata information regarding the sdk.Config type
+  ConfigurationDescriptor configuration = 4;
+  // query_services provides metadata information regarding the available queriable endpoints
+  QueryServicesDescriptor query_services = 5;
+  // tx provides metadata information regarding how to send transactions to the given application
+  TxDescriptor tx = 6;
+}
+
+// TxDescriptor describes the accepted transaction type
+message TxDescriptor {
+  // fullname is the protobuf fullname of the raw transaction type (for instance the tx.Tx type)
+  string fullname = 1;
+  // authn provides information on how to authenticate a transaction in offline mode
+  AuthnDescriptor authn = 2;
+  // msgs lists the accepted application messages (sdk.ServiceMsg, sdk.Msg)
+  // NOTE: not to be confused with proto.Message types
+  repeated MsgDescriptor msgs = 3;
+}
+
+// AuthnDescriptor provides information on how to sign transactions without relying
+// on the online RPCs GetTxMetadata and CombineUnsignedTxAndSignatures
+message AuthnDescriptor {
+  // sign_modes defines the supported signature algorithm
+  repeated SigningModeDescriptor sign_modes = 1;
+}
+
+// SigningModeDescriptor provides information on a signing flow of the application
+// NOTE(fdymylja): here we could go as far as providing an entire flow on how
+// to sign a message given a SigningModeDescriptor, but it's better to think about
+// this another time
+message SigningModeDescriptor {
+  // name defines the unique name of the signing mode
+  string name = 1;
```

Comment:
Can we instead reuse the `SIGN_MODE` enum directly here?

Reply:
Do we want to import into a reflection server a dependency on base?

Reply:
So I think there's a scoping issue here. I would say this should actually be in `cosmos.tx.v1.beta1.Service`. Maybe it's enough to describe what the tx service URL is and then we can get the sign modes from the tx service so you don't need to import.

Reply:
I do agree with what you're saying. But at the same time this would need the reflection client to rely not only on reflection and auth but on the tx service too. Uhm... I don't know

Reply:
I always understood "reflection" as "a proto service returning something about the proto files themselves". So strictly speaking, every field in this file should be a proto fullname string. With this mindset, I would also put this info in the TxService.

For example, here the TxDescriptor would only return proto fullnames:
- `fullname string` -> return `cosmos.tx.v1beta1.Tx`
- `service_name string` -> return `cosmos.tx.v1beta1.Service`
- `repeated MsgDescriptor msgs` -> return all the Msg Services fullnames

and other normal services can return whatever they want.

It seems that this PR understands "reflection" as "proto reflection + any chain config that's not queryable from state". Which is fine, but yeah, creates scoping issues as whether `chain-id`/`codecs`/`sign_modes`/`bech32` belong to normal services or reflection services.